source/common/filesystem/io_file_error.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/api/io_error.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Filesystem {
+
+class IoFileError : public Api::IoError {
+public:
+  explicit IoFileError(int sys_errno) : errno_(sys_errno) {}
+
+  ~IoFileError() override {}
+
+private:
+  IoErrorCode errorCode() const override;
```

Comment:
For these overrides, please add a comment that indicates what they implement. For an example of these comments, see https://github.com/envoyproxy/envoy/blob/master/source/common/access_log/access_log_formatter.h#L106

Reply:
Done.